### PR TITLE
Debug handler

### PR DESCRIPTION
**Pull request by damemi**
Depends on https://github.com/openshift/library-go/pull/321

Adds call to handler function to serve resource controller sync rules

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -148,6 +148,10 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 
 	configUpgradeableController := unsupportedconfigoverridescontroller.NewUnsupportedConfigOverridesController(operatorClient, ctx.EventRecorder)
 
+	if ctx.Server != nil {
+		ctx.Server.Handler.NonGoRestfulMux.HandleFunc("/syncRules", resourceSyncController.ServeSyncRules)
```
Comment (deads2k) [inline pkg/operator/starter.go]: This looks off.  

`/debug/controllers/resourcesync`

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -148,6 +148,10 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 
 	configUpgradeableController := unsupportedconfigoverridescontroller.NewUnsupportedConfigOverridesController(operatorClient, ctx.EventRecorder)
 
+	if ctx.Server != nil {
+		ctx.Server.Handler.NonGoRestfulMux.Handle("/debug/controllers/resourcesync", debugHandler)
```
Comment (deads2k) [inline pkg/operator/starter.go]: I suspect that we'll want `handleprefix` in the end, but ok.

**Comment by damemi** [comment]
ping @deads2k is this good to merge for now?

File: pkg/operator/resourcesynccontroller/resourcesynccontroller.go [hunk 3]
```diff
@@ -28,26 +30,26 @@ func NewResourceSyncController(
...
 	if err := resourceSyncController.SyncConfigMap(
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespace, Name: "aggregator-client-ca"},
 		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.GlobalMachineSpecifiedConfigNamespace, Name: "kube-apiserver-aggregator-client-ca"},
 	); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return resourceSyncController, nil
+	return resourceSyncController, resourcesynccontroller.NewDebugHandler(resourceSyncController), nil
```
Comment (sttts) [inline pkg/operator/resourcesynccontroller/resourcesynccontroller.go]: why do we need a return value for the handler if it is only built from the already returned `resourceSyncController` and `NewDebugHandler`?

**Comment by damemi** [comment]
@sttts based on feedback from David we decided to explicitly return the
debug handler (see
https://github.com/openshift/library-go/pull/321#issuecomment-476762957).
Am I misinterpreting something?

**Comment by damemi** [comment]
@sttts the other option is importing both `resourcesynccontroller` packages from this operator and from library-go into `starter.go`, which seems unnecessary since it would only be imported for this function

File: pkg/operator/starter.go (hunk 2 above)
Comment (sttts) [reply pkg/operator/starter.go]: Can ctx.Server be nil?

**Pull request merged**